packages/squiggle-lang/src/rescript/GenericDist/GenericDist_Types.res

```diff
@@ -0,0 +1,92 @@
+type genericDist = [
+  | #PointSet(PointSetTypes.pointSetDist)
+  | #SampleSet(array<float>)
+  | #Symbolic(SymbolicDistTypes.symbolicDist)
+]
+
+type error =
+  | NotYetImplemented
+  | ImpossiblePath
+  | DistributionVerticalShiftIsInvalid
+  | Other(string)
+
+module Operation = {
+  type direction = [
+    | #Algebraic
+    | #Pointwise
+  ]
+
+  type arithmeticOperation = [
```

Comment:
I see no reason for it to be polymorphic, but we can ignore for now and just make the task of doing https://github.com/QURIresearch/squiggle/issues/29 larger when the day comes that we do it.

Reply:
this comment could be in more places on the PR.

Reply:
Yea; polymorphic is easier to start with (less typing), but often less ideal.

Reply:
I agree, and I'm confused as to why this conversation is resolved, this variant is still polymorphic?

Reply:
I think resolved because it was decided to kick the can down the road. my first comment in the thread just said `29` would be larger. That seems fine with both of us.